File: Napakalaki/src/napakalaki/PruebaNapakalaki.java
    package napakalaki;

import java.util.List;
import java.util.stream.Collectors;

/**
 *
 * @author alexnp
 */
public class PruebaNapakalaki {
    
    static CardDealer dealer = CardDealer.getInstance();

    static List<Monster> monstruos = dealer.unusedMonsters;

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        
        dealer.initCards();
        
        monstruosConNivelMayorADiez().stream().forEach(System.out::println); 
        monstruosConMalRolloNiveles().stream().forEach(System.out::println);
        monstruosConBuenRolloDeNivelUno().stream().forEach(System.out::println);
        monstruosConsulta4(TreasureKind.ONEHAND).stream().forEach(System.out::println);
        monstruosConsulta4(TreasureKind.BOTHHANDS).stream().forEach(System.out::println);
        monstruosConsulta4(TreasureKind.HELMET).stream().forEach(System.out::println);
        monstruosConsulta4(TreasureKind.SHOES).stream().forEach(System.out::println);
        monstruosConsulta4(TreasureKind.ARMOR).stream().forEach(System.out::println);
        
        
    }
    
    public static List<Monster> monstruosConNivelMayorADiez() {
        return monstruos.stream()
                .filter(m -> m.getCombatLevel() > 10)
                .collect(Collectors.toList());
    }
    
    public static List<Monster> monstruosConMalRolloNiveles() {
        return monstruos.stream()
                .filter(m ->
                    !m.getBc().isDeath()
                        && m.getBc().getnVisibleTreasures() == 0
                        && m.getBc().getnHiddenTreasures() == 0
                        && m.getBc().getSpecificVisibleTreasure().isEmpty()
                        && m.getBc().getSpecificHiddenTreasure().isEmpty()
                )
                .collect(Collectors.toList());
    }
    
    public static List<Monster> monstruosConBuenRolloDeNivelUno() {
        return monstruos.stream()
                .filter(m -> m.getPrize().getLevel() > 1)
                .collect(Collectors.toList());
    }
    
    public static List<Monster> monstruosConsulta4(TreasureKind treasureKind) {
        return monstruos.stream()
                .filter(m ->
                        m.getBc().getSpecificVisibleTreasure().contains(treasureKind)
                        || m.getBc().getSpecificHiddenTreasure().contains(treasureKind)
                )
                .collect(Collectors.toList());
    }
}

File: Napakalaki-rb/pkg/Napakalaki-rb-0.0.1/lib/PruebaNapakalaki.rb
#encoding: utf-8

require_relative 'TreasureKind.rb'
require_relative 'Monster.rb'
require_relative 'CardDealer.rb'

module PruebaNapakalaki
  
  def self.monstruos_con_nivel_mayor_a_diez
    @@monstruos.select { |m| m.combatLevel > 10 }
  end
  
  def self.monstruos_con_mal_rollo_niveles
    @@monstruos.select { |m| 
      m.bc.nVisibleTreasures == 0 and
        m.bc.nHiddenTreasures == 0 and
        m.bc.specificVisibleTreasures.empty? and
        m.bc.specificHiddenTreasures.empty?
    }
  end
  
  def self.monstruos_con_mal_rollo_de_nivel_uno
    @@monstruos.select { |m|
      m.prize.level > 1
    }
  end
  
  def self.monstruos_consulta_4(treasure_kind)
    @@monstruos.select { |m|
      m.bc.specificVisibleTreasures.include?(treasure_kind) or
        m.bc.specificHiddenTreasures.include?(treasure_kind)
    }
  end

  def self.main
    puts monstruos_consulta_4(TreasureKind::ARMOR)
    puts monstruos_con_nivel_mayor_a_diez
    puts monstruos_con_mal_rollo_niveles
    puts monstruos_con_mal_rollo_de_nivel_uno
    puts monstruos_consulta_4(TreasureKind::ONEHAND)
    puts monstruos_consulta_4(TreasureKind::BOTHHANDS)
    puts monstruos_consulta_4(TreasureKind::HELMET)
    puts monstruos_consulta_4(TreasureKind::SHOES)
    puts monstruos_consulta_4(TreasureKind::ARMOR)
  end
  
  #start init
  @@dealer = CardDealer.instance
  @@dealer.init_cards
  @@monstruos = @@dealer.unused_monsters
  #end init
end

PruebaNapakalaki::main